docs/madr/decisions/076-naming-internal-envoy-resources.md

```diff
@@ -0,0 +1,319 @@
+In MeshProxyPatch you can use `name` matcher to match a resource name but if we change the name of the resource, it will not match anymore.
+The feature will be behind a feature flag so that nothing breaks unexpectedly but migration of MeshProxyPatch resources will be required.
+
+### As a Kuma developer I want to rename resources that are coming from outside Kuma
+
+This is either not possible or very hard.
+Envoy can connect to many xDS servers to get configuration from.
+If we integrate with systems like Spire (that exposes its own SDS server) we will not be able to rename resources that are coming from there.
+
+See:
+- https://github.com/spiffe/spire/blob/f687bf21e812a4bf027d88136032fc46497e7fe0/pkg/agent/endpoints/sdsv3/handler.go#L395
+- https://github.com/spiffe/spire/blob/f6a11a0ae03353b3444007ecaa8c81b34931bb49/pkg/agent/endpoints/sdsv3/handler.go#L282
+- https://github.com/spiffe/spire/blob/f6a11a0ae03353b3444007ecaa8c81b34931bb49/pkg/agent/endpoints/sdsv3/handler.go#L291
+- https://github.com/spiffe/spire/blob/f6a11a0ae03353b3444007ecaa8c81b34931bb49/pkg/agent/endpoints/sdsv3/handler.go#L312
+
+## Design
+
+### Resource naming
+
+#### 1. User resources
+
+Names for these resources are defined in [Defining and migrating to consistent naming for non-system Envoy resources and stats](./077-migrating-to-consistent-and-well-defined-naming-for-non-system-envoy-resources-and-stats.md).
+
+#### 2. System resources that can be tracked back to user resource
+
+When possible, use `system_<kri>`. It should be justified when designing why a user resource is system and not just `<kri>`.
+For example, when a resource isn't part of regular service-to-service traffic path, is shared across the system (like Kong Mesh's `MeshGlobalRateLimit` service), or simply needs to be easy to exclude from metrics collection.
+
+There are other cases where a resource is the result of merging multiple resources together (MeshTrafficPermission RBAC filters, MeshPassthrough, etc.).
+This is out of scope and requires a separate MADR.
+
+#### 3. System resources that are not related to user resource
+
+When adding such resource the name should be explicit enough to help anyone understand where this is coming from.
+The name should be namespaced so that it groups similar resources together and avoids collisions with other system resources.
+
+For example dynamic config fetcher creates one listener and one route per object type.
+We will namespace it as `system_dynamicconfig` and the DNS route will be `system_dynamicconfig_dns` and MeshMetric route will be `system_dynamicconfig_meshmetric`.
+
+If any related configuration option or annotation name exists it should be taken into account.
+
+#### 4. Resource contextual to the Dataplane
+
+These resources are defined in [Defining and migrating to consistent naming for non-system Envoy resources and stats](./077-migrating-to-consistent-and-well-defined-naming-for-non-system-envoy-resources-and-stats.md).
```

Comment:
Could you please add here these resources have `self_` pattern? It'd be nice to have all resource categories and their prefixes in one place and this MADR is the best candidate as it defines all the categories

Reply:
@lahabana suggested to have another document that summarizes all the categories in one place.